accounts-db/benches/read_only_accounts_cache.rs

```diff
@@ -0,0 +1,325 @@
+                            }
+                        }
+                    })
+                    .unwrap()
+            })
+            .collect::<Vec<_>>();
+
+        // Spawn the writer threads in the background.
+        let slot = 1;
+        let writer_handles = (0..*num_readers_writers)
+            .map(|i| {
+                let stop_threads = Arc::clone(&stop_threads);
+                let cache = Arc::clone(&cache);
+                let accounts = accounts.clone();
+
+                Builder::new()
+                    .name(format!("writer{i:02}"))
+                    .spawn({
+                        move || {
+                            // Continuously write to already existing pubkeys.
+                            let mut rng = SmallRng::seed_from_u64(100_u64.saturating_add(i as u64));
+                            while !stop_threads.load(Ordering::Relaxed) {
+                                let (pubkey, account) = accounts.choose(&mut rng).unwrap();
+                                cache.store(*pubkey, slot, account.clone());
+                            }
+                        }
+                    })
+                    .unwrap()
+            })
+            .collect::<Vec<_>>();
+
+        group.bench_function(BenchmarkId::new("store", num_readers_writers), |b| {
+            b.iter_custom(|iters| {
+                let mut total_time = Duration::new(0, 0);
+
+                for (pubkey, account) in accounts.iter().cycle().take(iters as usize) {
+                    // Measure only stores.
+                    let start = Instant::now();
+                    cache.store(*pubkey, slot, account.clone());
+                    total_time = total_time.saturating_add(start.elapsed());
+                }
+                total_time
+            })
+        });
+        group.bench_function(BenchmarkId::new("load", num_readers_writers), |b| {
+            b.iter_custom(|iters| {
+                let start = Instant::now();
+                for (pubkey, _) in accounts.iter().cycle().take(iters as usize) {
+                    black_box(cache.load(*pubkey, slot));
+                }
+
+                start.elapsed()
+            })
+        });
+
+        stop_threads.store(true, Ordering::Relaxed);
+        for reader_handle in reader_handles {
+            reader_handle.join().unwrap();
+        }
+        for writer_handle in writer_handles {
+            writer_handle.join().unwrap();
+        }
+    }
+}
+
+/// Benchmarks the read-only cache eviction mechanism. It does so by performing
+/// multithreaded reads and writes on a full cache. Each write triggers
+/// eviction. Background reads add more contention.
+fn bench_read_only_accounts_cache_eviction(
+    c: &mut Criterion,
+    group_name: &str,
+    max_data_size_lo: usize,
+    max_data_size_hi: usize,
+) {
+    // Prepare initial accounts, enough of the to fill up the cache.
+    let accounts: Vec<_> = utils::accounts_with_size_limit(
```

Comment:
> Looks like `accounts_with_size_limit()` will not exceed the passed-in size limit, so `accounts` here won't actually _completely_ fill up the cache. Is that right? Is that actually an issue since we are immediately doing some stores?

One more store is enough, so yeah, I would say it's not that important.

That said, I can try to think of some way to add one more element to the iterator to actually exceed the limit.

> (This is why I thought we had the 2x (or 10x) impl before.)

Yes, that was my intention, then I realized it's an overkill.

Reply:
I ended up doing 2x, to make sure that background write threads sometimes write an account which is not currently in the cache. Left a comment about that.